File: lib/redux_local.dart
@JS()
@JsMap('bower_components/polymer-redux/dist/polymer-redux.js')
library polymer_element.redux_local;

import 'package:js/js.dart';
import 'package:polymer_element/polymer_element.dart';
import 'package:polymerize_common/map.dart';
import 'package:polymer_element/redux.dart' as redux;

typedef Reducer(state, ReduxAction action);

@JS()
@anonymous
class ReduxAction<X> {
  external String get type;
  external X get detail;
  external factory ReduxAction({String type, X detail});
}

redux.Action<X> asAction<X>(ReduxAction<X> a) => new redux.Action<X>(a);

@BowerImport(
    ref: 'polymer-dart/polymer-redux#v2.0.1',
    import: "polymer-redux/polymer-redux-local.html",
    name: 'polymer-redux')
@JS("PolymerReduxLocal")
abstract class ReduxLocalBehavior {
  external get store;
  external set store(v);
  external dispatch(ReduxAction action);
}

File: lib/redux_local.ts
/** Library asset:polymer_element/lib/redux_local.dart */
import {is,isNot,equals} from "@dart2ts/dart/_common";
import {defaultConstructor,namedConstructor,namedFactory,defaultFactory,DartClass,Implements,With,op,Op,OperatorMethods,DartClassAnnotation,DartMethodAnnotation,DartPropertyAnnotation,Abstract,AbstractProperty,int,bool,double,Omit} from "@dart2ts/dart/utils";
import * as _common from "@dart2ts/dart/_common";
import * as core from "@dart2ts/dart/core";
import * as async from "@dart2ts/dart/async";
import * as lib3 from "./polymerize_js";
import * as lib4 from "./redux";

export var asAction : <X>(a : {type:string,detail?:any}) => lib4.Action<X> = <X>(a : {type:string,detail?:any}) : lib4.Action<X> =>  {
    return new lib4.Action<X>(a);
};
export class properties {
}
